server/common/src/main/java/org/eclipse/lsp/cobol/common/processor/ProcessingContext.java

```diff
@@ -39,18 +40,24 @@ public class ProcessingContext {
     private final CompilerDirectiveContext compilerDirectiveContext;
     private final Map<String, JsonElement> dialectsConfig;
 
+    private final LinkedList<ProgramNode> currentProgramNodeStack = new LinkedList<>();
+
     public ProcessingContext(List<SyntaxError> errors, VariableAccumulator variableAccumulator, Map<String, JsonElement> dialectsConfig) {
         this(errors, variableAccumulator, new CompilerDirectiveContext(), dialectsConfig);
     }
 
+    public ProgramNode getCurrentProgramNode() {
+        return currentProgramNodeStack.peek();
+    }
+
     /**
      * Register node type processor
      *
      * @param processorDesc Processor descriptor.
      */
     public void register(ProcessorDescription processorDesc) {
         processors
-                .computeIfAbsent(processorDesc.getPhase(), v -> new LinkedHashMap<>())
+                .computeIfAbsent(processorDesc.getPhase(), v -> new HashMap<>())
```

Comment:
Based on the original usage pattern I was actually wondering if this should be just an ArrayList.